cosmos/airflow/graph.py

```diff
@@ -643,6 +643,8 @@ def build_airflow_graph(
     detached_from_parent: dict[str, list[DbtNode]] = defaultdict(list)
     identify_detached_nodes(nodes, render_config, detached_nodes, detached_from_parent)
 
+    virtualenv_dir = task_args.pop("virtualenv_dir", None)
```

Comment:
Would this cause any issues for virtual env execution mode?

Reply:
Why are we popping this here..?

Reply:
Added a check before poping